articles/tutorials/building_2d_games/10_input_management/index.md

```diff
@@ -0,0 +1,966 @@
+{
+    switch (button)
+    {
+        case MouseButton.Left:
+            return CurrentState.LeftButton == ButtonState.Released && PreviousState.LeftButton == ButtonState.Pressed;
+        case MouseButton.Middle:
+            return CurrentState.MiddleButton == ButtonState.Released && PreviousState.MiddleButton == ButtonState.Pressed;
+        case MouseButton.Right:
+            return CurrentState.RightButton == ButtonState.Released && PreviousState.RightButton == ButtonState.Pressed;
+        case MouseButton.XButton1:
+            return CurrentState.XButton1 == ButtonState.Released && PreviousState.XButton1 == ButtonState.Pressed;
+        case MouseButton.XButton2:
+            return CurrentState.XButton2 == ButtonState.Released && PreviousState.XButton2 == ButtonState.Pressed;
+        default:
+            return false;
+    }
+}
+```
+
+These methods serve two distinct purposes. For checking continuous states:
+
+
+- `IsKeyDown`: Returns true as long as a key is being held down.
+- `IsKeyUp`: Returns true as long as a key is not being pressed.
+
+And for detecting state changes:
+- `WasKeyJustPressed`: Returns true only on the frame when a key changes from up-to-down.
+- `WasKeyJustReleased`: Returns true only on the frame when a key changes from down-to-up.
+
+> [!NOTE]
+> Each method uses a switch statement to check the appropriate button property from the [**MouseState**](xref:Microsoft.Xna.Framework.Input.MouseState) based on which `MouseButton` enum value is provided. This provides a consistent API while handling the different button properties internally.
+
+Finally, we need a method to handle setting the cursor position:
+
+```cs
+/// <summary>
+/// Sets the current position of the mouse cursor in screen space and updates the CurrentState with the new position.
+/// </summary>
+/// <param name="x">The x-coordinate location of the mouse cursor in screen space.</param>
+/// <param name="y">The y-coordinate location of the mouse cursor in screen space.</param>
+public void SetPosition(int x, int y)
+{
+    Mouse.SetPosition(x, y);
+    CurrentState = new MouseState(
+        x,
+        y,
+        CurrentState.ScrollWheelValue,
+        CurrentState.LeftButton,
+        CurrentState.MiddleButton,
+        CurrentState.RightButton,
+        CurrentState.XButton1,
+        CurrentState.XButton2
+    );
+}
+```
+
+> [!TIP]
+> Notice that after setting the position, we immediately update the `CurrentState`. This ensures our state tracking remains accurate even when manually moving the cursor.
+
+That's it for the `MouseInfo` class, next we'll move onto gamepad input.
+
+## The GamePadInfo Class
+
+To manage gamepad input effectively, we need to track both current and previous states, as well as provide easy access to the thumbstick values, trigger values, and button states. The `GamePadInfo` class will encapsulate all of this functionality, making it easier to:
+
+- Track current and previous mouse states.
```

Comment:
Should also track "is connected", it is mandatory for console titles to properly manage gamepad connection/disconnection.
But it is is also a good idea for PC's too, as well as using isConnected as to whether to bother getting data for a controller.